src/mtev_conf.c

```diff
@@ -94,6 +96,114 @@ static xmlDocPtr master_config = NULL;
 static int config_include_cnt = -1;
 static int backingstore_include_cnt = -1;
 
+struct param_entry {
+  char *name;
+  char *xpath;
+  mtev_param_type_t ptype;
+  void *memory;
+  mtev_param_validator_t validate;
+};
+
+
+mtev_boolean
+mtev_conf_register_global_param(const char *name, const char *xpath,
+                                mtev_param_type_t ptype, void *mem,
+                                mtev_param_validator_t validate) {
+  struct param_entry *p = calloc(1, sizeof(*p));
+  p->name = strdup(name);
+  if(xpath) p->xpath = strdup(xpath);
+  p->ptype = ptype;
+  p->memory = mem;
+  p->validate = validate;
+  if(mtev_hash_store(&global_param_sets, p->name, strlen(p->name), p))
+    return mtev_true;
+  free(p->name);
+  free(p->xpath);
+  free(p);
+  return mtev_false;
+}
+
+static mtev_boolean
+mtev_conf_update_global_param(const char *name, const char *value,
+                              mtev_boolean *running, mtev_boolean *config) {
+  void *vp;
+  struct param_entry *p;
+  char *endptr;
+  mtev_boolean b;
+  float f;
+  double d;
+  if(running) *running = mtev_false;
+  if(config) *config = mtev_false;
+  if(!mtev_hash_retrieve(&global_param_sets, name, strlen(name), &vp))
+    return mtev_false;
+  p = vp;
+
+  switch(p->ptype) {
+    case MTEV_PARAM_BOOLEAN:
+      if(!strcasecmp(value, "on") || !strcasecmp(value, "true")) {
+        b = mtev_true;
+        if(p->validate && !p->validate(p->ptype, &b)) return mtev_true;
+        *((mtev_boolean *)p->memory) = b;
+        if(running) *running = mtev_true;
+      }
+      else if(!strcasecmp(value, "off") || !strcasecmp(value, "false")) {
+        b = mtev_true;
+        if(p->validate && !p->validate(p->ptype, &b)) return mtev_true;
+        *((mtev_boolean *)p->memory) = b;
+        if(running) *running = mtev_true;
+      }
+      else {
+        return mtev_true;
```

Comment:
The function is supposed to return false if the parameter name is not found; true otherwise.

The two other params state whether the actions were successful.